test/test_transformers.py

```diff
@@ -1340,8 +1340,8 @@ def test_invalid_last_dim_stride(self, device, kernel: SDPBackend):
                     q, k, v, None, 0.0, False))
 
     @onlyCUDA
-    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_ATTENTION, "Does not flash_attention fused scaled dot product attention")
-    @parametrize("kernel", PLATFORM_SPECIFIC_SDPA)
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FLASH_ATTENTION, "Does not flash_attention fused scaled dot product attention")
```

Comment:
I think this isn't quite right...
Do we even need `get_platform_specific_sdpa` any more? Or since we added the rocm specific logic to their definitions can we keep the test parameterizations as is and only add `skipifROCM` where needed?

Reply:
Although the benefits of `get_platform_specific_sdpa` are not very clear, which is due to we are focusing to get the infrastructure PR merged ASAP without enabling all tests, but in some cases we do.

The current UT has an assumption: "If a platform supports Flash Attention, it must support Memory efficient attention as well". However this is not true for ROCM (and other platforms). 
`get_platform_specific_sdpa` is introduced to address this problem in a single place, and it will replace more `@parametrize("kernel")` when we are enabling more tests.

Additionally there is another benefit. If a new type of attention backend were added, only this function needs updates.

Reply:
I do agree that this is a nice pattern, especially since it is likely that backends could be added. 

I don't love that we can't return empty lists to parametrize with the assumption that if the list is empty the test will be skipped entirely, would you mind creating an issue for this? This feels like a potentially nice feature.

Reply:
> I do agree that this is a nice pattern, especially since it is likely that backends could be added.
> 
> I don't love that we can't return empty lists to parametrize with the assumption that if the list is empty the test will be skipped entirely, would you mind creating an issue for this? This feels like a potentially nice feature.

Submitted as https://github.com/pytorch/pytorch/issues/115202